### PR TITLE
in_array('defer', 'async') with helper function

### PR DESCRIPTION
**Pull request by kt-12**
Trac ticket: 

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by 10upsimon** [comment]
@kt-12 as per my Slack comms, while I am not necessarily _against_ the naming of `is_delayed_strategy`, delaying something is grammatically not the same as deferring it, so I've suggested potentially two alternatives here:

- `is_postponed_strategy`
- `is_optimized_strategy`

But I am keen for @joemcgill and @adamsilverstein's take on this, @felixarntz too.

File: src/wp-includes/class-wp-scripts.php [hunk 1]
```diff
@@ -875,6 +876,18 @@ private function get_intended_strategy( $handle ) {
 		return isset( $script_args['strategy'] ) ? $script_args['strategy'] : false;
 	}
 
+	/**
+	 * Check if the strategy passed is one of the delayed strategies.
+	 *
+	 * @param string $strategy The script handle.
+	 * @return string|bool Strategy set during script registration. False if none was set.
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-scripts.php]: @kt-12 In which case we get string value in return type? [in_array](https://www.php.net/manual/en/function.in-array.php) return only bool value.

File: src/wp-includes/class-wp-scripts.php [hunk 2]
```diff
@@ -875,6 +876,18 @@ private function get_intended_strategy( $handle ) {
 		return isset( $script_args['strategy'] ) ? $script_args['strategy'] : false;
 	}
 
+	/**
+	 * Check if the strategy passed is one of the delayed strategies.
+	 *
+	 * @param string $strategy The strategy to check.
+	 * @return bool True if $strategy is one of the delayed strategy.
+	 */
+	private function is_delayed_strategy( $strategy ) {
```
Comment (adamsilverstein) [inline src/wp-includes/class-wp-scripts.php]: for core, convention is prefixing function names with `_`

Comment (felixarntz) [reply src/wp-includes/class-wp-scripts.php]: @adamsilverstein I don't think so? Private methods should never be prefixed, and even for privately annotated functions there is no clear convention. When I started contributing to core years ago, I got feedback from several seasoned core team members that I should only put `@access private` on the function, but not underscore-prefix it. Opinions appear to differ on that, but at least for class methods we should not use prefixes.

Comment (joemcgill) [reply src/wp-includes/class-wp-scripts.php]: For global functions, we use the underscore. For private class methods, I believe @felixarntz is correct.

Also: naming things is hard. What about `is_non_blocking_strategy`? Seems that is generally the intent of that method anyway, right?

**Review by joemcgill — COMMENTED** [review]
Looking good to me, other than landing on a better name for this method. Left a suggestion in the comments.

File: src/wp-includes/class-wp-scripts.php (hunk 2 above)
Comment (kt-12) [reply src/wp-includes/class-wp-scripts.php]: @joemcgill that name looks perfect to me. I'll rename it.

File: src/wp-includes/class-wp-scripts.php [hunk 3]
```diff
@@ -882,6 +883,18 @@ private function get_intended_strategy( $handle ) {
 		return isset( $script_args['strategy'] ) ? $script_args['strategy'] : false;
 	}
 
+	/**
+	 * Check if the strategy passed is one of the delayed strategies.
```
Comment (joemcgill) [inline src/wp-includes/class-wp-scripts.php]:
```suggestion
	 * Check if the strategy passed is a non-blocking strategy.
```

**Review by joemcgill — APPROVED** [review]
Small doc suggestion, but this looks good to me 👍🏻

**Review by 10upsimon — APPROVED** [review]

File: src/wp-includes/class-wp-scripts.php (hunk 3 above)
Comment (10upsimon) [reply src/wp-includes/class-wp-scripts.php]: @joemcgill @kt-12 I've addressed this in a recent commit, I am going to merge this PR.

**Pull request merged**